src/Tokenizer/Tokens.php

```diff
@@ -1424,6 +1424,10 @@ private function unregisterFoundToken($token)
         }
 
         --$this->foundTokenKinds[$tokenKind];
+
+        if (0 === $this->foundTokenKinds[$tokenKind]) {
```

Comment:
I treat `empty` for arrays as combination of `isset` and `count`, but twice faster than calling both of them manually.

In your proposal, to not have the need to call both, `isset` and `count`, we have extra condition + unset in `unregisterFoundToken` method, that is called damn often.
Before merging the PR, we need to investigate potential performance impact.

Reply:
On the other hand we gain with replacing 3 times `empty` with `isset`. Any hint how can we reasonably investigate performance change?